CMSgov/design-system
# [WNMGDS-2553] Sketch plugin: Add inheritance logic for text styles

## Summary

WNMGDS-2553

Adds inheritance logic for text styles based on the naming conventions of the keys. Consider the following(incomplete) heading tokens for the core theme:
```json
"typography-heading": {
    "__font-family": "inherit",
    "__font-weight": 700,
    "-sm__font-weight": 400,
    "-md__font-weight": 700,
    "-lg__font-weight": 700,
    "-sm__font-size": ".875rem",
    "-md__font-size": "1rem",
    "-lg__font-size": "1.125rem",
}
```

The code that previously generated the text styles would have created the following text styles (sticking with JSON format for consistency, even though these would be converted to Sketch's internal representation of text styles):

```json
"typography-heading": {
    "font-family": "inherit",
    "font-weight": 700,
},

"typography-heading-sm": {
    "font-weight": 400,
    "font-size": ".875rem",
},

"typography-heading-md": {
    "font-weight": 700,
    "font-size": "1rem",
},

"typography-heading-lg": {
    "font-weight": 700,
    "font-size": "1.125rem",
}
```

The problem with this—alluded to in [#2777](https://github.com/CMSgov/design-system/pull/2777)—is that assigning text styles in Sketch is _all or nothing_. You can't apply both the "typography-heading" and "typography-heading-sm" to text in Sketch, so those styles need to contain all the information needed to render that text. In this case, we want the `-sm`, `-md`, and `-lg` styles to inherit from the base styles. That's where this new logic comes in.

The new code uses the naming conventions in a group of component tokens to determine a parent style name, and the text styles can therefore inherit properties from the parent. (And technically everything inherits from a "base" text style called `defaultTextStyle`, which is equivalent to the text-related styles we apply to the `body` element in the CSS normalization rules.) The generated text styles with the new logic would look more like this:

```json
"typography-heading": {
    "font-family": "Open Sans",
    "font-weight": 700,
},

"typography-heading-sm": {
    "font-family": "Open Sans",
    "font-weight": 400,
    "font-size": ".875rem",
},

"typography-heading-md": {
    "font-family": "Open Sans",
    "font-weight": 700,
    "font-size": "1rem",
},

"typography-heading-lg": {
    "font-family": "Open Sans",
    "font-weight": 700,
    "font-size": "1.125rem",
}
```

## How to test

1. Check out this branch
2. Run `yarn install && yarn build` in your terminal, where the current directory is the root of the repository
3. Run `cd packages/design-system-tokens/sketch/cmsds-token-importer`
4.  Run `yarn install && yarn build`
5. Open a Sketch document
6. Click on the "Plugins" menu and select "CMSDS Color Token Importer" (or double-click on the `packages/design-system-tokens/sketch/cmsds-token-importer/cmsds-token-importer.sketchplugin` file to install the plugin if it's not in the menu)
7. Select a theme
8. Click on the "Show components" button and then the "Show text styles" button
9. The plugin will have created "components" and "typography" folders (lowercase). You can browse the styles there.

**This change is most noticeable if you select the "Medicare" theme and look at the "components" > "label" styles. Previously the inverse styles did not have the right font family or weight.**

## Checklist
